src/main.cpp

```diff
@@ -232,6 +219,15 @@ bool KodiAdaptiveStream::download(const char* urlPointer, const char* rangeHeade
     xbmc->CURLAddOption(file, XFILE::CURL_OPTION_HEADER, "Range", rangeHeader);
   xbmc->CURLAddOption(file, XFILE::CURL_OPTION_HEADER, "Connection", "keep-alive");
   xbmc->CURLAddOption(file, XFILE::CURL_OPTION_PROTOCOL, "acceptencoding", "gzip, deflate");
+  if (strcmp(segmentHeaders, "") != 0)
+  {
+    std::vector<std::string> header, headers = split(segmentHeaders, '&');
```

Comment:
If the string is encoded.. would you not be splitting on %26?

Maybe best to decode the entire string before this first split?
Then you don't need the decode in the for loop